### PR TITLE
Ignore Malformed ASGs

### PR DESCRIPTION
**Pull request by jconstance-amplify**

### CODE REVIEW

**Review by aoskotsky-amplify — APPROVED** [review]

**Review by vlutsyuk-amplify — APPROVED** [review]

File: disco_aws_automation/disco_autoscale.py [hunk 1]
```diff
@@ -54,17 +54,21 @@ def get_launch_config_name(self, hostclass):
...
     def _filter_by_environment(self, items):
         """Filters autoscaling groups and launch configs by environment"""
-        return [
-            item for item in items
-            if item.name.startswith("{0}_".format(self.environment_name))
-        ]
+        for item in items:
+            try:
+                if item.name.startswith("{0}_".format(self.environment_name)):
+                    yield item
```
Comment (skim-amplify) [inline disco_aws_automation/disco_autoscale.py]: are we sure that the returned items will only be iterated over once?

File: disco_aws_automation/disco_autoscale.py [hunk 2]
```diff
@@ -54,17 +54,21 @@ def get_launch_config_name(self, hostclass):
...
     def _filter_instance_by_environment(self, items):
         """Filter instances by environment via their group_name"""
-        return [
-            item for item in items
-            if item.group_name.startswith("{0}_".format(self.environment_name))
-        ]
+        for item in items:
+            try:
+                if item.group_name.startswith("{0}_".format(self.environment_name)):
+                    yield item
```
Comment (skim-amplify) [inline disco_aws_automation/disco_autoscale.py]: are we sure that the returned items will only be iterated over once?

**Review by skim-amplify — APPROVED** [review]
if you're sure that the returned items is iterated only once then :+1: 
if not, then just append to list and return that at the end

File: disco_aws_automation/disco_autoscale.py (hunk 1 above)
Comment (jconstance-amplify) [reply disco_aws_automation/disco_autoscale.py]: Looking at the usage, looks like yes.

**Pull request merged**